lib/sycamore/sycamore/docset.py

```diff
@@ -1531,29 +1530,45 @@ def execute(self, **kwargs) -> None:
         from sycamore.executor import Execution
         from sycamore.materialize import Materialize
 
-        if isinstance(self.plan, Materialize) and self.plan._reliability is not None:
-            mrr = self.plan._reliability
-
-            assert self.plan.child().last_node_materialize(), "first step must be materialize step"
+        if isinstance(self.plan, Materialize):
+            mrr = None
```

Comment:
Can you refactor this to have the reliable execution logic in the same file as MRR?  Then the code can be:
```
if maybe_execute_reliably(self):
   pass # execution already ran
else:
    for doc in Execution(self.context).execute_iter(self.plan, **kwargs):
          pass

with the expected return True/False in maybe_execute_reliably
```

This function is borderline too long and is mixing reliable and unreliable execution.
I think when you refactor, you can also split the function. The first half is figuring out if running into reliability mode so can end with:
```
if mrr is None:
   return False
else
   execute_reliably(docset, mrr, destinationPath)
```